### ISSUE TITLE
laser_scan topic no data published

### ISSUE THREAD

**Issue by kennyS99** [ouster-lidar/ouster-ros#237]
**Describe the bug**
I've encountered a bug where the /ouster/scan topic does not seem to publish any data.
Here is the screenshot of the RVIZ that only subscribes to the topic of /scan:
![Screenshot from 2023-10-17 20-23-59](https://github.com/ouster-lidar/ouster-ros/assets/34148678/0a1acf8c-ed9d-4dee-8a8c-b1e0d78e9194)

I checked the pointClound2 publish, and it is working. here is the screenshot.
![Screenshot from 2023-10-17 20-27-12](https://github.com/ouster-lidar/ouster-ros/assets/34148678/80d18145-97d5-43b6-8aea-c47b74140433)

Also, I attempted to run  ros2 echo topic /ouster/scan. and the output is here:
[ouster.txt](https://github.com/ouster-lidar/ouster-ros/files/12928896/ouster.txt)

It seems like there is no data that has been published here. Additionally, I am pretty sure that the variable of proc_mask has been added with SCAN. cuz when I run ros2 topic list, I can see the /ouster/scan will come with the output.      
 declare_parameter<std::string>("proc_mask", "IMU|IMG|PCL|SCAN");

**To Reproduce**
Steps to reproduce the behaviour (steps below are just an example):
ros2 launch ouster_ros driver.launch.py

**Platform (please complete the following information):**
- Ouster Sensor? OS-1
- Ouster Firmware Version? firmware 2.0
- ROS version/distro? ros2 rolling
- Operating System? \ Linux with ubuntu docker 22.04
- Machine Architecture? \ x64
- git commit hash (if not the latest). i think that I am using the default one.
If there is any possible solution gonna be very helpful for me. I appreciate your help.

**Comment by Samahu** [comment]
Did you set the value of `scan_ring` parameter to a value other than zero? It is not required but I am wondering that you sensor could be setting on a table and most of the lower beam hit very close such that the sensor is returning zeros for the range value. So if you have an OSx-64 sensor you typically want to set `scan_ring` to 32, and to 64 if you had an OSx-128 sensor.

Hope this helps!

**Comment by kennyS99** [comment]
Hello Samahu,
Thank you for your prompt response. I attempted the solution you provided today. I'm uncertain about the OSx-64 reference, so I experimented with various values for the scan_ring parameter in the YAML file, including 32 and 64.

Result:
The output from the /ouster/scan topic appears to be functioning as expected now. Please refer to the attached file for details.
[ouster.txt](https://github.com/ouster-lidar/ouster-ros/files/12998863/ouster.txt)

However, I'm facing an issue with rviz2. Even after adding a scan module to subscribe to the /scan topic, it's not being visualized correctly in rviz2.
![Screenshot from 2023-10-18 15-44-44](https://github.com/ouster-lidar/ouster-ros/assets/34148678/29c65ab5-7a74-4fe6-89e7-1e0ac5bb5ea3)

The pointClound2 topic has been working as a reference before.

**Comment by Samahu** [comment]
In PR #203 (which is still WIP) I have added a LidarScan object to the rviz config.. You could double check if there is anything essential different than yours. One thing to keep an eye on is that in order for the `/ouster/scan` topic to appear correctly, RVIZ configuration should match with the topic QoS setting.

**Comment by kennyS99** [comment]
Thank you for your continuous support. After updating my Rviz configuration file, I can now visualize the 2D LaserScan in Rviz.

However, I've encountered an alignment issue that you mentioned previously. Specifically, I believe I'm experiencing a similar issue to what was described in #196. In my case, the offset between the pointCloud and the laser scan is significant, almost 180 degrees. This discrepancy is quite concerning.
![Screenshot](https://github.com/ouster-lidar/ouster-ros/assets/34148678/350621c9-ceb7-4999-a888-63895432924b)

Are you still working on this problem? I'm keen to know if other users are experiencing the same discrepancy, or if it's unique to my setup.

Best regards,

**Comment by lowellausen** [comment]
Hi @kennyS99 ! Another user here. We have faced that issue as well; the thing is that the LaserScan data is, currently, not actually being transformed into it's reference frame, whereas the PointCLoud data is being transformed. 
I have recently opened an issue and a not-so-beautiful  PR about it (https://github.com/ouster-lidar/ouster-ros/issues/235 https://github.com/ouster-lidar/ouster-ros/pull/236). And this is the original issue about it: https://github.com/ouster-lidar/ouster-ros/issues/186

There's an even uglier way to workaround this and that is setting, in you config file, your `lidar_frame` to be the same as your `sensor_frame`. That surely messes up your tf tree a bit but they will align :joy:
